cocos/particle/renderer/particle-system-renderer-cpu.ts

```diff
@@ -136,20 +137,24 @@ const _matInsInfo: IMaterialInstanceInfo = {
     subModelIdx: 0,
 };
 
+// TODO: we should not use this type, should use a uniform array type instead.
+// Tracking issue: https://github.com/cocos/cocos-engine/issues/15553
+export type PVData = [Vec3, Vec3, Vec3, Vec3, number, Vec3 | null, null];
```

Comment:
What does it mean?

Reply:
this means that this is an array type, and its element type is uncertain

for example 
```ts
let a: PVData;
a[0] // Vec3 type
a[4]  // number type
```

Reply:
> and its element type is uncertain

I think it is defined in the PVData. As `PVData[0]` should be `Vec3`, `PVData[1]` should be `Vec3`, and so on.

Reply:
yes, because it was type of `any` before, it is very wierd to implement like this

we should not introduce different element types in an array, this issue should be solved here https://github.com/cocos/cocos-engine/issues/15553